gilmour.go

```diff
@@ -523,16 +536,14 @@ func isExpired(timeout time.Duration, startTime time.Time) bool {
 	return time.Now().After(startTime.Add(timeout))
 }
 
-func try(r RetryConf, fn func() error) (err error) {
+func try(g *Gilmour, fn func() error) (err error) {
 	startTime := time.Now()
 	for {
 		err = fn()
-		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
+		if isExpired(g.getRetryTimeout(), startTime) || !isNetworkError(err) || err == nil {
```

Comment:
@meson10 My first question is when you say 
```basically checking if the timeout was exceeded after the timeout happens.```
what timeout do you mean the timeout for the retries or the timeout of the request. I would assume that in case there are problems connecting to the backend the big O would be dependent more on the cost of retries then actually publishing to redis.  I am assuming that you want us to check the timeout of the request before making the request. That makes complete sense to me but if you have seen the mail from aditya, We should follow the rule of "Follow the rule of least surprise". I believe the least surprise is if we check for the request timeout as well. I will leave the final decision to you @meson10 . @mohandutt134